__test__/api/comments/updateSingleCommentUpVotes.test.js

```diff
@@ -0,0 +1,276 @@
+        expect(comment.downVotes.length).toEqual(
+          oldCommentWithDownVote.numOfDownVotes
+        );
+        expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+          oldCommentWithDownVote.numOfVotes
+        );
+      }
+    );
+
+    // Run test matchers to verify that the comment votes addition produced the correct success response.
+    const addRes = await request
+      .patch(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: ownerIdUpdate,
+      });
+
+    expect(addRes.status).toEqual(200);
+    expect(addRes.body.status).toEqual("success");
+    expect(addRes.body.data).toEqual({
+      commentId: oldCommentWithDownVote.commentId,
+      numOfVotes: oldCommentWithDownVote.numOfVotes,
+      numOfUpVotes: oldCommentWithDownVote.numOfUpVotes + 1,
+      numOfDownVotes: oldCommentWithDownVote.numOfDownVotes - 1,
+    });
+
+    // Run test matchers to verify that the comment votes have been added in the database.
+    await CommentModel.findById(oldCommentWithDownVote.commentId).then(
+      (comment) => {
+        expect(comment).toBeTruthy();
+        expect(comment.upVotes.length).toEqual(
+          oldCommentWithDownVote.numOfUpVotes + 1
+        );
+        expect(comment.downVotes.length).toEqual(
+          oldCommentWithDownVote.numOfDownVotes - 1
+        );
+        expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+          oldCommentWithDownVote.numOfVotes
+        );
+      }
+    );
+  });
+
+  it("Should remove a upvote from a comment (toggle)", async () => {
+    const url = `/v1/comments/${oldCommentWithUpVote.commentId}/votes/upvote`;
+    const bearerToken = `bearer ${global.appToken}`;
+    const ownerIdUpdate = "voter@gmail.com";
+
+    // Run test matchers to verify that the comment votes have not been removed in the database.
+    await CommentModel.findById(oldCommentWithUpVote.commentId).then(
+      (comment) => {
+        expect(comment).toBeTruthy();
+        expect(comment.upVotes.length).toEqual(
+          oldCommentWithUpVote.numOfUpVotes
+        );
+        expect(comment.downVotes.length).toEqual(
+          oldCommentWithUpVote.numOfDownVotes
+        );
+        expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+          oldCommentWithUpVote.numOfVotes
+        );
+      }
+    );
+
+    // Run test matchers to verify that the comment votes removal produced the correct success response.
+    const removeRes = await request
+      .patch(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: ownerIdUpdate,
+      });
+
+    expect(removeRes.status).toEqual(200);
+    expect(removeRes.body.status).toEqual("success");
+    expect(removeRes.body.data).toEqual({
+      commentId: oldCommentWithUpVote.commentId,
+      numOfVotes: oldCommentWithUpVote.numOfVotes - 1,
+      numOfUpVotes: oldCommentWithUpVote.numOfUpVotes,
+      numOfDownVotes: oldCommentWithUpVote.numOfDownVotes - 1,
```

Comment:
Since you are removing an upvote, shouldn't you be expecting the `numOfUpVotes` to be reduced by one. Not the `numOfDownVotes`.